### PR TITLE
Refactor logic to backend

### PR DESCRIPTION
**Pull request by tjomson**
#se include a summary of the change and which issue it fixes. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- The logic handling how to draw the graph is now done on the backend.
- Added separate endpoints that just return the nodes and the links
- When a node is clicked, it now sends a request to get all of the info about that particular server
- This way we only send the information that is necessary at any given time, and the backend does all of the processing, so the frontend only has to display what it receives.

Any thoughts on this structure?
I want to do something with SignalR, to maybe make the communication better, but i have not looked into it.

I added the classes Node and Link that cut down versions of Server and Link. They are inner classes for now, but should they be added to the models-folder? Maybe in a sub-folder so that it is clear that they are not like Connection, Gateway, etc.

# How was this tested?

Manually

# Type of change

Please check all boxes that apply to this change below.

| Type                     | Applies ✅|
| ------------------------ | :-------: |
| Backend                  |   ✅        |
| Frontend                 |   ✅        |
| Github/Ci/Admin          |           |
| Documentation required   |           |

# Checklist

Please follow the checklist below and check all completed items as you work on the PR.

I have..
- [x] performed a self-review of my own code
- [ ] commented my code, particularly in hard-to-understand areas
- [ ] written documentation
- [ ] added automated tests
- [ ] merged dependencies to `develop

### CODE REVIEW

File: src/components/Graph.vue [hunk 1]
```diff
@@ -96,9 +95,9 @@ export default {
       .attr('fill', d => d.ntv_error ? '#f00' : '#000')
       .call(drag(simulation))
       .on('click', (d, i) => {
-        console.log(i.server_id)
-        console.log(d)
-        console.log(i)
+        axios.get('https://localhost:5001/varz/' + i.server_id).then(a => {
```
Comment (hojelse) [inline src/components/Graph.vue:98]: We might want to abstract all get requests to a class, so we can mock/stub the backend. Having to spin up the backend every time is getting tedious 😅

File: backend/Controller.cs [hunk 2]
```diff
@@ -54,5 +79,69 @@ public ActionResult<IEnumerable<Leaf>> GetLeafz()
             return Program.leafs;
         }
 
+        public static void processData()
```
Comment (khabalevskyi) [inline backend/Controller.cs:82]: Pascal case for public members? :)
```suggestion
        public static void ProcessData() 
```

File: src/components/Graph.vue (hunk 1 above)
Comment (tjomson) [reply src/components/Graph.vue:98]: That is true. The domain should probably be configurable in some way, so we don't need to have the specific URL in the code, and also be able to change it for testing and such

File: backend/Program.cs [hunk 3]
```diff
@@ -118,6 +118,8 @@ private static void Main(string[] args)
                 }
             });
 
+            Controller.processData();
```
Comment (khabalevskyi) [inline backend/Program.cs:121]:
```suggestion
            Controller.ProcessData();
```

Comment (tjomson) [reply backend/Program.cs:121]: Oh, i have not tried this "commit suggestion" before. Apparently it doesn't rename references

**Review by pizzaluc — APPROVED** [review]
Looks good!

**Pull request merged**